continuous-integration/requirements.Rmd

```diff
@@ -0,0 +1,28 @@
+---
+title: "Requirements for continuous integration"
+maintainer: "Thierry Onkelinx"
+output: html_document
+---
+
+# Must have
+
+- start automatically when pushing a git repository on Github or Bitbucket
+- start automatically when creating a pull request on Github or Bitbucket
+- able to use a Docker image to setup the environment in which CI runs
+- able to set environment variables
+- able to check code style with lintr and report the results back to the SCM
+- able to check code coverage with covr and report the results back to the SCM
```

Comment:
Concerning the 2 previous steps, I do think we should divide requirements on the general level (agnostic of the language) and how this is translated towards requirements on the programming language level (R, Python and Java). 

So, we do want to include for any language:
* support for unit testing  (cfr. `testthat` in R and  `pytest` in python)
* code syntax check`lint`  (cfr. `lintr` in R and `pylint` in python)
* test coverage evaluation (cfr. `covr` in R and `coverage` in python) -> could be linked to something like [coveralls](https://coveralls.io/)

This should be properly integrated within the CI setup (as easy as possible)